### PR TITLE
Fixed some Bugs and Code Smells that SonarCloud reported

### PR DESCRIPTION
**Pull request by apple00juice**
Mainly removing public or adding final at some locations.

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java [hunk 1]
```diff
@@ -88,7 +88,6 @@ public ILobby createLobby(String token, String lobbyName, Visibility visibility,
         try {
             newLobby = lobbyManager.createLobby(lobbyName, visibility);
         }catch(SmallestIdNotCreatable e){
-           e.printStackTrace();
            String errorMessage = "The server could not generate a unique id";
            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, errorMessage);
```
Comment (hilalcomak) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:92]: throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, errorMessage, e);

If we add the cause "e" to the constructor we can trace the exception back to the origin.

**Review by hilalcomak — APPROVED** [review]
Added a comment about Exception traceability.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java (hunk 1 above)
Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:92]: Good point, but the question is how does the json return object handle this. It is possible that we then leak implementation details through the API. I will quickly test what the API returns, or do you know it by chance ?

Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:92]: _id\"; nested exception is ch.uzh.ifi.hase.soprafs22.exceptions.SmallestIdNotCreatable\r\n\tat_

_Caused by: ch.uzh.ifi.hase.soprafs22.exceptions.SmallestIdNotCreatable\r\n\tat ch.uzh.ifi.hase.soprafs22.lobby.LobbyManager.createLobby(LobbyManager.java:64)\r\n\tat ch.uzh.ifi.hase.soprafs22.service.LobbyService.createLobby(LobbyService.java:89)\r\n\t... 63 more\r\n",_

These two things get added to the trace field in the return json object, I don't think that we need to hide them and we get better traceability from it as you said. 

Thus I would add your change if you agree

Comment (paolovic) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:92]: Agree at first sight.
For the sake of naming, do you want to add an Exception at SmallestIdNotCreatable so that it becomes SmallestIdNotCreatableException?

Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:92]: Yea, that sounds better. I will definitely do that.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 public class LobbyManager implements ILobbyManager {
 
-    private volatile static LobbyManager uniqueInstance;
+    private static final LobbyManager uniqueInstance = new LobbyManager();
```
Comment (paolovic) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java:23]: Was that a code smell?
Before you had the (almost) textbook like Singleton applied, why did you change it?

**Review by paolovic — APPROVED** [review]
I'm just curious about the Singleton thing, but I'm sure you have a good reason for it.
Merge if you want @apple00juice

File: src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java (hunk 2 above)
Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java:23]: Yes, sonarCloud said, if I remember correctly that volatile only works with atomic types or something like that. I then read online that the method before is not completely thread save. The new way is the thread safest option as far as I know and read but you always create an instance even if you might not need it. In our case that shouldn't be a problem as the lobby manager is crucial and needs to be generated anyways. The old way would probably also work but one less smell I guess (but maybe it will grumble about that too 😢)

Comment (paolovic) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java:23]: Ah I remember, that was one of the drawbacks of Singletons, it's not thread safe.
Although it promises by design a single instance, multiple threads could create multiple Singletons.
